Database/Constraint/sp_add_or_update_locker.sql

```diff
@@ -0,0 +1,55 @@
+﻿/*==============================================================*/
+/* DBMS name:		FestiBase									*/
+/* PDM version 4												*/	
+/* Robert Verkerk												*/
+/* Stored procedure													*/
+/*==============================================================*/
+USE FestiBase
+GO
+
+DROP PROCEDURE IF EXISTS sp_add_or_update_locker
+GO
+CREATE PROCEDURE sp_add_or_update_locker
+	@insert BIT,
+	@locker_number INT = NULL,
+	@festival_company_number INT
+AS
+BEGIN
+	BEGIN TRY
+		-- Inser or update start
+		IF (@insert = 1)
+		BEGIN
+			INSERT INTO LOCKER (festival_company_number) VALUES (@festival_company_number)
+			-- insert code
+		END
+		ELSE 
+		BEGIN
+			IF (@locker_number = null)
+			BEGIN
+				;THROW 50000, 'You must supply a surrogate key with an update statement.', 1
+			END
```

Comment:
ELSE IF EXISTS (SELECT * FROM locker WHERE locker_number = @locker_number)
BEGIN
;THROW 50000, 'This locker does not exist', 1
END